guides/Advanced SettingGateway/UnderstandingSchemaFolders.md

```diff
@@ -0,0 +1,164 @@
+# Understanding Schema
+
+A schema works like a diagram or a blueprint, in SettingGateway, the schema defines the keys present in the configuration for a specific gateway. This feature serves multiple purposes:
+
+1. Define what keys does the {@link Gateway} manage and their properties.
+1. Define what type the keys must hold.
+1. Define the SQL schema when using a SQL database.
+1. Speed up performance when iterating over keys.
+
+## Adding keys
+
+Adding keys with the schema is like adding a piece into a box, but you can also have boxes inside another boxes. That being said, you get the box you want to modify and insert the new pieces or boxes into it. The methods to achieve that are {@link SchemaFolder#addKey} to add pieces (keys) and {@link SchemaFolder#addFolder} to add boxes (more folders).
```

Comment:
You can also have boxes inside ***other*** boxes

To add pieces (keys) and {@link SchemaFolder#addFolder} to add boxes (~more~folders)